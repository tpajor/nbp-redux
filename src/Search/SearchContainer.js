import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { addCard, deleteCard, showDetails, getCurrencyRequest } from '../Actions';

import Search from './Search';
import Card from './Card';
import CardList from './CardList';

import './SearchContainer.css';

const SearchContainer = (props) => {
  return (
    <div className={'col-4 SearchView'}>
      <Search getCurrency={props.getCurrency}
        searchError={props.searchViewData.searchError}
        searchErrorMessage={props.searchViewData.searchErrorMessage}
      />
      {(props.searchViewData.temporaryCard !== null) ?
        <Card cardData={props.searchViewData.temporaryCard}
          addCard={props.addCard}
          showDetails={props.showDetails}
          isTemporary={true}
        /> :
        <br />
      }
      <CardList cards={props.searchViewData.cards}
        deleteCard={props.deleteCard}
        showDetails={props.showDetails}
      />
    </div>
  );
};

const mapDispatchToProps = {
  addCard,
  deleteCard,
  showDetails,
  getCurrency: getCurrencyRequest,
};

SearchContainer.propTypes = {
  searchViewData: PropTypes.object,
};

export default connect(null, mapDispatchToProps)(SearchContainer);