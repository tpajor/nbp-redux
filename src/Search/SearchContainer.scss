.SearchView {
  border: 2px solid gray;
  border-style: none solid none none;
  min-height: 100vh;
}