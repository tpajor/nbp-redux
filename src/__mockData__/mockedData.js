const card = { id: 'someId', code: 'USD', currency: 'dolar amerykański', rates: [{ ask: 3.3, bid: 3.4 }]};
const card2 = { id: 'someId2', code: 'USD', currency: 'dolar amerykański', rates: { ask: 3.3, bid: 3.4 }};

const search = {
  cards: [card, card2],
  temporaryCard: card,
  searchError: false,
  addCardError: true,
  addCardErrorMessage: 'error',
  currenciesTable: ['usd', 'chf', 'aud'],
  currencyCodeToViewInDetail: 'usd',
};

const user = { userName: 'User', pass: 'pass' };
const user2 = { userName: 'User2', pass: 'pass2'};

export const login = {
  signingError: true,
  signingErrorMessage: 'error',
  signedIn: true,
  userSignedIn: 'User',
  users: [user, user2]
};

const lastRates = [
  { ask: 3.5983, bid: 3.5271, effectiveDate: "2018-05-07", no: "087/C/NBP/2018" },
  { ask: 3.6088, bid: 3.5374, effectiveDate: "2018-05-08", no: "088/C/NBP/2018" }
];

const detail = {
  lastRates: lastRates,
  detailsInputError: false,
  detailsInputErrorMessage: '',
};

export const initialState = { search, login, detail };

export const searchViewData = {
  cards: search.cards,
  temporaryCard: search.temporaryCard,
  searchError: search.searchError,
  signedIn: login.signedIn,
  userSignedIn: login.userSignedIn,
  addCardError: search.addCardError,
  addCardErrorMessage: search.addCardErrorMessage,
  currenciesTable: search.currenciesTable,
};

export const detailViewData = {
  currencyCodeToViewInDetail: search.currencyCodeToViewInDetail,
  lastRates: detail.lastRates,
  detailsInputError: detail.detailsInputError,
  detailsInputErrorMessage: detail.detailsInputErrorMessage,
  signedIn: login.signedIn,
};