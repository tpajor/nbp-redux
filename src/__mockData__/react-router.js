const Link = () => {
  return jest.fn();
}

export default Link;