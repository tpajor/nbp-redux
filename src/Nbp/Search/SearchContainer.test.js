import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SearchContainer } from './SearchContainer';
import { searchViewData } from '../../__mockData__/mockedData';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    deleteCard: jest.fn(),
    addCard: jest.fn(),
    showDetails: jest.fn(),
    getCurrency: jest.fn(),
    getCurrenciesTable: jest.fn(),
    searchViewData: searchViewData
  };

  const enzymeWrapper = shallow(<SearchContainer {...props} />);

  return {
    enzymeWrapper,
    props
  };
};

describe('SearchContainer component', () => {
  it('should render self and subcomponents with their props', () => {
    const { enzymeWrapper, props } = setup();

    expect(enzymeWrapper.length).toBe(1);

    expect(enzymeWrapper.find('div').length).toBe(1);
    expect(enzymeWrapper.find('div').hasClass('col-4 SearchView')).toBe(true);

    expect(enzymeWrapper.find('Search').length).toBe(1);
    expect(typeof enzymeWrapper.find('Search').props().getCurrenciesTable).toBe('function');
    expect(typeof enzymeWrapper.find('Search').props().getCurrency).toBe('function');
    expect(enzymeWrapper.find('Search').props().searchError).toEqual(searchViewData.searchError);
    expect(enzymeWrapper.find('Search').props().currenciesTable).toEqual(searchViewData.currenciesTable);
    expect(enzymeWrapper.find('CardList').length).toBe(1);
    expect(typeof enzymeWrapper.find('CardList').props().deleteCard).toBe('function');
    expect(typeof enzymeWrapper.find('CardList').props().showDetails).toBe('function');
    expect(enzymeWrapper.find('CardList').props().cards).toEqual(searchViewData.cards);
    expect(enzymeWrapper.find('CardList').props().signedIn).toEqual(searchViewData.signedIn);
    expect(enzymeWrapper.find('CardList').props().userSignedIn).toEqual(searchViewData.userSignedIn);

    enzymeWrapper.setProps({ ...props, searchViewData: { ...searchViewData, userSignedIn: 'User' } });
    expect(enzymeWrapper.find('h5').length).toBe(1);
    enzymeWrapper.setProps({ ...props, searchViewData: { ...searchViewData, userSignedIn: '' } });
    expect(enzymeWrapper.find('h5').length).toBe(0);
    enzymeWrapper.setProps({ ...props });

    enzymeWrapper.setProps({ ...props, searchViewData: { ...searchViewData, addCardError: true } });
    expect(enzymeWrapper.find('p').length).toBe(1);
    expect(enzymeWrapper.find('p').text()).toEqual(searchViewData.addCardErrorMessage);
    expect(enzymeWrapper.find('p').children('br').length).toBe(1);
    enzymeWrapper.setProps({ ...props, searchViewData: { ...searchViewData, addCardError: false } });
    expect(enzymeWrapper.find('p').length).toBe(0);
    enzymeWrapper.setProps({ ...props });

    enzymeWrapper.setProps({ ...props, searchViewData: { ...searchViewData, temporaryCard: null } });
    expect(enzymeWrapper.find('Card').length).toBe(0);
    enzymeWrapper.setProps({ ...props, searchViewData: { ...searchViewData, temporaryCard: searchViewData.temporaryCard } });
    expect(enzymeWrapper.find('Card').length).toBe(1);
    expect(typeof enzymeWrapper.find('Card').props().addCard).toBe('function');
    expect(typeof enzymeWrapper.find('Card').props().showDetails).toBe('function');
    expect(enzymeWrapper.find('Card').props().cardData).toEqual(searchViewData.temporaryCard);
    expect(enzymeWrapper.find('Card').props().signedIn).toEqual(searchViewData.signedIn);
    expect(enzymeWrapper.find('Card').props().userSignedIn).toEqual(searchViewData.userSignedIn);
    expect(enzymeWrapper.find('Card').props().isTemporary).toBe(true);
    expect(enzymeWrapper.find('Card').props().cards).toEqual(searchViewData.cards);
    });

  it('should render correctly', () => {
    const { enzymeWrapper } = setup();
    
    expect(enzymeWrapper).toMatchSnapshot();
  });
});