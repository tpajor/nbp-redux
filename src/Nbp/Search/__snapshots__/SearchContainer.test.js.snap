// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addCard={[Function]}
    deleteCard={[Function]}
    getCurrenciesTable={[Function]}
    getCurrency={[Function]}
    searchViewData={
        Object {
            "addCardError": true,
            "addCardErrorMessage": "error",
            "cards": Array [
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId2",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
            ],
            "currenciesTable": Array [
              "usd",
              "chf",
              "aud",
            ],
            "searchError": false,
            "signedIn": true,
            "temporaryCard": Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            "userSignedIn": "User",
          }
    }
    showDetails={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Witaj, User
</h5>,
        <Search
          currenciesTable={
                    Array [
                              "usd",
                              "chf",
                              "aud",
                            ]
          }
          getCurrenciesTable={[Function]}
          getCurrency={[Function]}
          searchError={false}
/>,
        <Card
          addCard={[Function]}
          cardData={
                    Object {
                              "code": "USD",
                              "currency": "dolar amerykański",
                              "id": "someId",
                              "rates": Array [
                                Object {
                                  "ask": 3.3,
                                  "bid": 3.4,
                                },
                              ],
                            }
          }
          cards={
                    Array [
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId",
                                "rates": Array [
                                  Object {
                                    "ask": 3.3,
                                    "bid": 3.4,
                                  },
                                ],
                              },
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId2",
                                "rates": Array [
                                  Object {
                                    "ask": 3.3,
                                    "bid": 3.4,
                                  },
                                ],
                              },
                            ]
          }
          isTemporary={true}
          showDetails={[Function]}
          signedIn={true}
          userSignedIn="User"
/>,
        <p
          style={
                    Object {
                              "color": "red",
                              "textAlign": "center",
                            }
          }
>
          <br />
          error
</p>,
        <CardList
          cards={
                    Array [
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId",
                                "rates": Array [
                                  Object {
                                    "ask": 3.3,
                                    "bid": 3.4,
                                  },
                                ],
                              },
                              Object {
                                "code": "USD",
                                "currency": "dolar amerykański",
                                "id": "someId2",
                                "rates": Array [
                                  Object {
                                    "ask": 3.3,
                                    "bid": 3.4,
                                  },
                                ],
                              },
                            ]
          }
          deleteCard={[Function]}
          showDetails={[Function]}
          signedIn={true}
          userSignedIn="User"
/>,
      ],
      "className": "col-4 SearchView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Witaj, User",
        },
        "ref": null,
        "rendered": "Witaj, User",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currenciesTable": Array [
            "usd",
            "chf",
            "aud",
          ],
          "getCurrenciesTable": [Function],
          "getCurrency": [Function],
          "searchError": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addCard": [Function],
          "cardData": Object {
            "code": "USD",
            "currency": "dolar amerykański",
            "id": "someId",
            "rates": Array [
              Object {
                "ask": 3.3,
                "bid": 3.4,
              },
            ],
          },
          "cards": Array [
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId2",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
          ],
          "isTemporary": true,
          "showDetails": [Function],
          "signedIn": true,
          "userSignedIn": "User",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <br />,
            "error",
          ],
          "style": Object {
            "color": "red",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "error",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": Array [
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId2",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
          ],
          "deleteCard": [Function],
          "showDetails": [Function],
          "signedIn": true,
          "userSignedIn": "User",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Witaj, User
</h5>,
          <Search
            currenciesTable={
                        Array [
                                    "usd",
                                    "chf",
                                    "aud",
                                  ]
            }
            getCurrenciesTable={[Function]}
            getCurrency={[Function]}
            searchError={false}
/>,
          <Card
            addCard={[Function]}
            cardData={
                        Object {
                                    "code": "USD",
                                    "currency": "dolar amerykański",
                                    "id": "someId",
                                    "rates": Array [
                                      Object {
                                        "ask": 3.3,
                                        "bid": 3.4,
                                      },
                                    ],
                                  }
            }
            cards={
                        Array [
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId",
                                      "rates": Array [
                                        Object {
                                          "ask": 3.3,
                                          "bid": 3.4,
                                        },
                                      ],
                                    },
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId2",
                                      "rates": Array [
                                        Object {
                                          "ask": 3.3,
                                          "bid": 3.4,
                                        },
                                      ],
                                    },
                                  ]
            }
            isTemporary={true}
            showDetails={[Function]}
            signedIn={true}
            userSignedIn="User"
/>,
          <p
            style={
                        Object {
                                    "color": "red",
                                    "textAlign": "center",
                                  }
            }
>
            <br />
            error
</p>,
          <CardList
            cards={
                        Array [
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId",
                                      "rates": Array [
                                        Object {
                                          "ask": 3.3,
                                          "bid": 3.4,
                                        },
                                      ],
                                    },
                                    Object {
                                      "code": "USD",
                                      "currency": "dolar amerykański",
                                      "id": "someId2",
                                      "rates": Array [
                                        Object {
                                          "ask": 3.3,
                                          "bid": 3.4,
                                        },
                                      ],
                                    },
                                  ]
            }
            deleteCard={[Function]}
            showDetails={[Function]}
            signedIn={true}
            userSignedIn="User"
/>,
        ],
        "className": "col-4 SearchView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Witaj, User",
          },
          "ref": null,
          "rendered": "Witaj, User",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currenciesTable": Array [
              "usd",
              "chf",
              "aud",
            ],
            "getCurrenciesTable": [Function],
            "getCurrency": [Function],
            "searchError": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addCard": [Function],
            "cardData": Object {
              "code": "USD",
              "currency": "dolar amerykański",
              "id": "someId",
              "rates": Array [
                Object {
                  "ask": 3.3,
                  "bid": 3.4,
                },
              ],
            },
            "cards": Array [
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId2",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
            ],
            "isTemporary": true,
            "showDetails": [Function],
            "signedIn": true,
            "userSignedIn": "User",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <br />,
              "error",
            ],
            "style": Object {
              "color": "red",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "error",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": Array [
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
              Object {
                "code": "USD",
                "currency": "dolar amerykański",
                "id": "someId2",
                "rates": Array [
                  Object {
                    "ask": 3.3,
                    "bid": 3.4,
                  },
                ],
              },
            ],
            "deleteCard": [Function],
            "showDetails": [Function],
            "signedIn": true,
            "userSignedIn": "User",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
